lib/mk-soxin-module.nix

```
@@ -34,5 +36,23 @@ recursiveUpdate
       if builtins.isString value then config.soxin.themes.${value}.${name}
       else value.${name};
   });
+
+  tool = optionalAttrs includeTool (mkOption {
+    type = with types; listOf [ str modules.tools.toolsModule ];
+    default = config.soxin.settings.tools;
+    description = "Tool to use for ${name}.";
+    apply = value:
+      if builtins.isString value then config.soxin.tools.${value}.${name}
+      else value.${name};
+  });
+
+  programmingLanguage = optionalAttrs includeprogrammingLanguage (mkOption {
+    type = with types; listOf [ str modules.programmingLanguages.programmingLanguagesModule ];
+    default = config.soxin.settings.programmingLanguages;
+    description = "Programming language to use for ${name}.";
+    apply = value:
+      if builtins.isString value then config.soxin.programmingLanguages.${value}.${name}
+      else value.${name};
```

```suggestion
    apply = value: map (v:
      if builtins.isString v then config.soxin.programmingLanguages.${v}.${name}
      else v.${name}
    ) value;
```
As it is now a list, you also have to modify the `apply` function.

---

thanks, this makes sense.